entity-framework/core/what-is-new/ef-core-5.0/whatsnew.md

```diff
@@ -14,6 +14,350 @@ This page does not duplicate the [plan for EF Core 5.0](xref:core/what-is-new/ef
+    "Name" TEXT NULL
+);
+
+INSERT INTO "ef_temp_Unicorns" ("Id", "Name")
+SELECT "Id", "Name"
+FROM Unicorns;
+
+PRAGMA foreign_keys = 0;
+
+DROP TABLE "Unicorns";
+
+ALTER TABLE "ef_temp_Unicorns" RENAME TO "Unicorns";
+
+PRAGMA foreign_keys = 1;
+```
+
+Notice that:
+* A temporary table is created with the desired schema for the new table
+* Data is copied from the current table into the temporary table
+* Foreign key enforcement is switched off
+* The current table is dropped
+* The temporary table is renamed to be the new table
+
+Documentation is tracked by issue [#2583](https://github.com/dotnet/EntityFramework.Docs/issues/2583).
+
+### Table-valued functions
+
+This feature was contributed from the community by [@pmiddleton](https://github.com/pmiddleton). Many thanks for the contribution!
+
+EF Core 5.0 includes first-class support for mapping .NET methods to table-valued functions (TVFs). These functions can then be used in LINQ queries where additional composition on the results of the function will also be translated to SQL.
+
+For example, consider this TVF defined in a SQL Server database:
+
+```sql
+create FUNCTION GetReports(@employeeId int)
+RETURNS @reports TABLE
+(
+	Name nvarchar(50) not null,
+	IsDeveloper bit not null
+)
+AS
+begin
+	WITH cteEmployees AS
+	(
+		SELECT id, name, managerId, isDeveloper
+		FROM employees
+		WHERE id = @employeeId
+		UNION ALL
+		SELECT e.id, e.name, e.managerId, e.isDeveloper
+		FROM employees e
+		INNER JOIN cteEmployees cteEmp ON cteEmp.id = e.ManagerId
+	)
+
+	insert into @reports
+	select name, isDeveloper
+	FROM cteEmployees
+	where id != @employeeId
+
+	return
+end
+```
+
+The EF Core model requires two entity types to use this TVF:
+* An `Employee` type that maps to the Employees table in the normal way
+* A `Report` type that matches the shape returned by the TVF
+
+```c#
+public class Employee
+{
+    public int Id { get; set; }
+    public string Name { get; set; }
+    public bool IsDeveloper { get; set; }
+
+    public int? ManagerId { get; set; }
+    public virtual Employee Manager { get; set; }
+}
+```
+
+```c#
+public class Report
+{
+    public string Name { get; set; }
+    public bool IsDeveloper { get; set; }
+}
+```
+
+These types must be included in the EF Core model:
+
+```c#
+modelBuilder.Entity<Employee>();
+modelBuilder.Entity(typeof(Report)).HasNoKey();
+```
+
+Notice that `Report` has no primary key and so must be configured as such.
+
+Finally, a .NET method must be mapped to the TVF in the database. This method can be defined on the DbContext using the new `FromExpression` method:
+
+```c#
+public IQueryable<Report> GetReports(int managerId)
+    => FromExpression(() => GetReports(managerId));
+```
+
+This method uses a parameter and return type that match the TVF defined above. The method is then added to the EF Core model in OnModelCreating:
+
+```c#
+modelBuilder.HasDbFunction(() => GetReports(0));
```

Comment:
```suggestion
modelBuilder.HasDbFunction(() => GetReports(default));
```

Reply:
I wondered about doing this. Probably still going to confuse some people. :-)